### PR TITLE
Update dependencies.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?

This pull request updates our dependencies, including updated [webpack config](https://github.com/DoSomething/webpack-config/pull/14), [Puck client](https://github.com/DoSomething/puck-client/pull/8), and some patched linting rules (yay for `getDerivedStateFromProps` moving back below constructor!)

### Any background context you want to provide?

I've clicked around a bit and everything seems to work well but may have missed something! Please give a shout if y'all see anything interesting that may have been caused by a dependency update.

### What are the relevant tickets/cards?

N/A

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [ ] Added screenshot to PR description of related front-end updates on **small** screens.
* [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
* [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by DFurnes** [comment]
Waiting on DoSomething/puck-client#9 to fix those failing Jest tests.

File: resources/assets/components/actions/VoterRegistrationAction/VoterRegistrationAction.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ const VoterRegistrationAction = props => {
             href={parsedLink}
             onClick={handleClick}
             target="_blank"
+            rel="noopener noreferrer"
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/components/actions/VoterRegistrationAction/VoterRegistrationAction.js]: Yeah, it's outrageous!

**Comment by DFurnes** [comment]
Alright, tests passing again & ready for review! 👓

File: package.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "private": true,
   "scripts": {
-    "test": "jest --watch",
-    "test:ci": "jest",
+    "test": "NODE_ENV=test && jest --watch",
+    "test:ci": "NODE_ENV=test && jest",
```
Comment (mendelB) [inline package.json]: What are the implications of setting `NODE_ENV` to `test` here? (Or is it just for conventional purposes for now?)

Comment (DFurnes) [reply package.json]: Ah! That was leftover from some testing, Jest actually sets that on it's own by default.

Comment (DFurnes) [reply package.json]: Removed in 12f806d.

**Pull request merged**